sound/soc/sof/ipc4-topology.c

```diff
@@ -923,8 +923,21 @@ static int sof_ipc4_widget_setup_comp_process(struct snd_sof_widget *swidget)
 	}
 
 	cfg_size = sizeof(struct sof_ipc4_base_module_cfg);
-	if (process->init_payload_format == INIT_PAYLOAD_WITH_OUTPUT_FMT)
+	switch (process->init_payload_format) {
+	case INIT_PAYLOAD_WITH_OUTPUT_FMT:
 		cfg_size += sizeof(struct sof_ipc4_audio_format);
+		break;
+	case INIT_PAYLOAD_WITH_BASE_CFG_EXT:
```

Comment:
I am confused, what would the initialization type from from the topology tokens? Is this not something hard-coded in the module code?